Fixes for linux build

pac85 opened the pull request:
This PR includes two commits. One works around an internal compiler error that occurs when trying to build with GCC.
The other commit simply adds the `+x` flags the the bash scripts so that users don't have to `chmod +x` them

DreamyCecil commented:
How did you manage to catch these errors? When I was building it for the first time, it never showed me any errors on printing functions, so I had go the "hard path" and [create weird workarounds to force them](https://github.com/DreamyCecil/Serious-Engine/blob/60e67d80f204a46bd1ef361f89b1f289532dafb2/Sources/Engine/SE_Config.h#L120), otherwise all printed text was displayed as random garbage.

I've fixed all the string conversions but I forgot about this constructor method with VA. I added `CTString::ConstData()` method for use in place of `const char *` conversions, so I'd like for you to use that instead.
Or I can fix it myself post-merge, along with other places like that.

pac85 commented:
> How did you manage to catch these errors? When I was building it for the first time, it never showed me any errors on printing functions, so I had go the "hard path" and [create weird workarounds to force them](https://github.com/DreamyCecil/Serious-Engine/blob/60e67d80f204a46bd1ef361f89b1f289532dafb2/Sources/Engine/SE_Config.h#L120), otherwise all printed text was displayed as random garbage.
> 

Mmm interesting, I get the following internal compiler error https://gist.github.com/pac85/fa45dbef79725318d1e53f13153a6eff

Running this version of gcc
```
$ gcc --version
gcc (GCC) 14.2.1 20240805
```

So this might have been a regression in GCC itself (I feel like this code should compile even if it causes UB at runtime)

> I've fixed all the string conversions but I forgot about this constructor method with VA. I added `CTString::ConstData()` method for use in place of `const char *` conversions, so I'd like for you to use that instead. Or I can fix it myself post-merge, along with other places like that.

Sure I'll update this PR

DreamyCecil commented:
Thanks.
